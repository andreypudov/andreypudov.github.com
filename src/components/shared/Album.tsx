import Model from '../../models/core/photographyAlbum';
import Photograph from '../../models/core/photograph';
import PhotographView from './Photograph';
import VideoView from './Video';

type AlbumProps = {
  album: Model;
};

function Album(props: AlbumProps) {
  return (
    <>
    {
      props.album.getMedia().map((media, index: number) =>
        {
          switch (Object.getPrototypeOf(media)) {
            case Photograph.prototype:
              return <PhotographView album = { props.album } index = { index } key = { index } />;
            default:
              return <VideoView album = { props.album } index = { index } title = { true } description = { true } key = { index } />;
        }})
    }
    </>
  );
}

export default Album;