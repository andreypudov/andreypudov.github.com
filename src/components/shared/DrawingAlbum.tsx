import Album from './../../models/core/photographyAlbum';
import AlbumView from '../shared/Album';
import Headline from './../shared/Headline';
import Layout from './../shared/Layout';

type DrawingAlbumProps = {
   album: Album;
 };

function DrawingAlbum({ album }: React.PropsWithChildren<DrawingAlbumProps>) {
  return (
    <Layout title = { album.getTitle() }>
      <div className = 'container'>
        <Headline album = { album } />
        <AlbumView album = { album } />
      </div>
    </Layout>
  );
}

export default DrawingAlbum;