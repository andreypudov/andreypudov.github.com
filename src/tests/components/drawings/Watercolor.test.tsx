import { MemoryRouter } from 'react-router-dom'
import DrawingAlbum from './../../../components/shared/DrawingAlbum';
import Album from './../../../models/drawings/watercolor';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HelmetProvider>
        <MemoryRouter>
          <DrawingAlbum album={Album} />
        </MemoryRouter>
      </HelmetProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});