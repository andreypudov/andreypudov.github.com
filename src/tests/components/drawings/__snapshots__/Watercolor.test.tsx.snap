// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="layout"
>
  <header>
    <div
      className="container"
    >
      <div
        className="navbar navbar-expand-md navbar-dark"
      >
        <div
          className="container-fluid"
        >
          <a
            className="navbar-brand mx-auto"
            href="/en/"
            onClick={[Function]}
          >
            Andrey Pudov
          </a>
          <button
            aria-controls="navbarMain"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-bs-target="#navbarMain"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-sm-center"
            id="navbarMain"
          >
            <ul
              className="navbar-nav"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/en/albums/"
                  onClick={[Function]}
                >
                  Albums
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/en/contents/"
                  onClick={[Function]}
                >
                  Contents
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/en/resume/"
                  onClick={[Function]}
                >
                  Resume
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/en/contact/"
                  onClick={[Function]}
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main>
    <div
      className="container"
    >
      <h1>
        Watercolor
        <small
          className="text-muted"
        >
          Sunday, September 25, 2022
        </small>
      </h1>
      <figure>
        <img
          alt="Watercolor - September 25, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-09-25.webp"
          title="September 25, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          September 25, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - July 20, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-07-20.webp"
          title="July 20, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          July 20, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - July 18, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-07-18.webp"
          title="July 18, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          July 18, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - Lost Friend / June 28, 2022"
          className="img-responsive gallery-image"
          src="/drawings/watercolor/2022-06-28.webp"
          title="Lost Friend / June 28, 2022"
        />
        <figcaption
          className="image-caption"
        >
          Lost Friend / June 28, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - Lisa / June 21, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-06-21.webp"
          title="Lisa / June 21, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          Lisa / June 21, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - May 31, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-05-31.webp"
          title="May 31, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          May 31, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - May 25, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-05-25.webp"
          title="May 25, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          May 25, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - March 29, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-03-29.webp"
          title="March 29, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          March 29, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - March 01, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-03-01.webp"
          title="March 01, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          March 01, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - February 28, 2022"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2022-02-28.webp"
          title="February 28, 2022"
        />
        <figcaption
          className="image-caption vertical"
        >
          February 28, 2022
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - May 05, 2021"
          className="img-responsive gallery-image"
          src="/drawings/watercolor/2021-05-24.webp"
          title="May 05, 2021"
        />
        <figcaption
          className="image-caption"
        >
          May 05, 2021
        </figcaption>
      </figure>
      <figure>
        <img
          alt="Watercolor - March 03, 2021"
          className="img-responsive gallery-image vertical"
          src="/drawings/watercolor/2021-03-30.webp"
          title="March 03, 2021"
        />
        <figcaption
          className="image-caption vertical"
        >
          March 03, 2021
        </figcaption>
      </figure>
    </div>
  </main>
  <footer>
    <div
      className="container"
    >
      <hr
        className="footer-divider"
      />
      <ul
        className="piped-links float-end"
      >
        <li
          className="first piped-links-short"
        >
          <a
            href="/en/resume/"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="first piped-links-long"
        >
          <a
            href="/en/resume/"
            onClick={[Function]}
          >
            About Andrey Pudov
          </a>
        </li>
        <li>
          <a
            href="/en/contact/"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
      <ul
        className="piped-links"
      >
        <p
          className="piped-links-long"
        >
          Copyright © 2023 Andrey Pudov. All rights reserved.
        </p>
        <p
          className="piped-links-short"
        >
          © 2023 Andrey Pudov. All rights reserved.
        </p>
      </ul>
    </div>
  </footer>
</div>
`;
